controllers/exclude_instances.go

```diff
@@ -40,9 +40,16 @@ func (e ExcludeInstances) Reconcile(r *FoundationDBClusterReconciler, context ct
 	}
 	defer adminClient.Close()
 
-	addresses := make([]string, 0, len(cluster.Status.PendingRemovals))
+	removalCount := 0
+	for _, processGroup := range cluster.Status.ProcessGroups {
```

Comment:
Instead of using a counter here we could create a new slice only containing the processGroups marked for removal, this way we don't have to traverse the processGroups twice.

Reply:
I think we're going to have to traverse it twice no matter what, because we want to ignore addresses that have already been excluded.

Reply:
That's right but we would traverse the complete list once and a second time the new list which contains only the removed once (with the assumption that in most cases the complete list will be much smaller than the list with the ProcessGroups to remove)

Reply:
But we can also optimise this in a next step.